File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/Go/s573993837_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var (
	L, R, top, bottom int
)

func main() {
	inputReader := bufio.NewReader(os.Stdin)
	outputWriter := bufio.NewWriter(os.Stdout)
	defer outputWriter.Flush()
	sc := NewScanner(inputReader)
	solve(sc, outputWriter)
}

func solve(sc *Scanner, pw *bufio.Writer) {
	t := 1
	for t > 0 {
		t--
		n := sc.NextInt64()
		m := sc.NextInt64()
		if n == 0 {
			fmt.Fprintln(pw, 0)
			continue
		}
		if n == m {
			fmt.Fprintln(pw, 0)
			continue
		}
		if n > m {
			n = m + n%m
			n -= m
			fmt.Fprintln(pw, min(m-n, n))
			continue
		}
		fmt.Fprintln(pw, min(m-n, n))
	}
}

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func swap(chrr []rune, i, j int) {
	chrr[i], chrr[j] = chrr[j], chrr[i]
}

func num(n int) int {
	a := 0
	for n > 0 {
		a += n & 1
		n >>= 1
	}
	return a
}

type Pair struct {
	u, v, i, val int
}

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	if n <= 3 {
		return true
	}
	if n%2 == 0 || n%3 == 0 {
		return false
	}
	for i := 5; i*i <= n; i += 6 {
		if n%i == 0 || n%(i+2) == 0 {
			return false
		}
	}
	return true
}

func gcd(a, b int64) int64 {
	if b == 0 {
		return a
	}
	if a > b {
		return gcd(b, a%b)
	}
	return gcd(a, b%a)
}

func fastPow(base, n, M int64) int64 {
	if n == 0 {
		return 1
	}
	if n == 1 {
		return base
	}
	halfn := fastPow(base, n/2, M)
	if n%2 == 0 {
		return (halfn * halfn) % M
	}
	return (((halfn * halfn) % M) * base) % M
}

func modInverse(n, M int64) int64 {
	return fastPow(n, M-2, M)
}

func feedArr(arr []int64, sc *Scanner) {
	for i := range arr {
		arr[i] = sc.NextInt64()
	}
}

func feedArrStr(arr []string, sc *Scanner) {
	for i := range arr {
		arr[i] = sc.Next()
	}
}

func printArr(arr []int) string {
	var sb strings.Builder
	for _, i := range arr {
		sb.WriteString(fmt.Sprintf("%d ", i))
	}
	return sb.String()
}

func printArrInt64(arr []int64) string {
	var sb strings.Builder
	for _, i := range arr {
		sb.WriteString(fmt.Sprintf("%d ", i))
	}
	return sb.String()
}

func printArrStr(arr []string) string {
	var sb strings.Builder
	for _, i := range arr {
		sb.WriteString(fmt.Sprintf("%s ", i))
	}
	return sb.String()
}

type Scanner struct {
	r *bufio.Reader
}

func NewScanner(r *bufio.Reader) *Scanner {
	return &Scanner{r}
}

func (s *Scanner) Next() string {
	var buf []byte
	for {
		b, err := s.r.ReadByte()
		if err != nil {
			panic(err)
		}
		if b == ' ' || b == '\n' {
			break
		}
		buf = append(buf, b)
	}
	return string(buf)
}

func (s *Scanner) NextInt64() int64 {
	i, err := strconv.ParseInt(s.Next(), 10, 64)
	if err != nil {
		panic(err)
	}
	return i
}

func (s *Scanner) NextInt() int {
	return int(s.NextInt64())
}

func (s *Scanner) NextFloat64() float64 {
	f, err := strconv.ParseFloat(s.Next(), 64)
	if err != nil {
		panic(err)
	}
	return f
}

func (s *Scanner) NextLine() string {
	var buf []byte
	for {
		b, err := s.r.ReadByte()
		if err != nil {
			panic(err)
		}
		if b == '\n' {
			break
		}
		buf = append(buf, b)
	}
	return string(buf)
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/direct/Java/C/s573993837_comments_removed.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

typedef struct {
    int u, v, i, val;
} Pair;

bool isPrime(int n) {
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 == 0 || n % 3 == 0) return false;
    for (int i = 5; i * i <= n; i += 6) {
        if (n % i == 0 || n % (i + 2) == 0) return false;
    }
    return true;
}

long gcd(long a, long b) {
    if (b == 0) return a;
    return a > b ? gcd(b, a % b) : gcd(a, b % a);
}

long fast_pow(long base, long n, long M) {
    if (n == 0) return 1;
    if (n == 1) return base;
    long halfn = fast_pow(base, n / 2, M);
    if (n % 2 == 0) return (halfn * halfn) % M;
    else return (((halfn * halfn) % M) * base) % M;
}

long modInverse(long n, long M) {
    return fast_pow(n, M - 2, M);
}

void feedArr(long *arr, int size) {
    for (int i = 0; i < size; i++) {
        scanf("%ld", &arr[i]);
    }
}

void feedArrInt(int *arr, int size) {
    for (int i = 0; i < size; i++) {
        scanf("%d", &arr[i]);
    }
}

void feedArrDouble(double *arr, int size) {
    for (int i = 0; i < size; i++) {
        scanf("%lf", &arr[i]);
    }
}

void feedArrString(char **arr, int size) {
    for (int i = 0; i < size; i++) {
        scanf("%s", arr[i]);
    }
}

char* printArr(int *arr, int size) {
    static char result[1000] = "";
    result[0] = '\0';
    for (int i = 0; i < size; i++) {
        char temp[10];
        sprintf(temp, "%d ", arr[i]);
        strcat(result, temp);
    }
    return result;
}

char* printArrLong(long *arr, int size) {
    static char result[1000] = "";
    result[0] = '\0';
    for (int i = 0; i < size; i++) {
        char temp[20];
        sprintf(temp, "%ld ", arr[i]);
        strcat(result, temp);
    }
    return result;
}

char* printArrString(char **arr, int size) {
    static char result[1000] = "";
    result[0] = '\0';
    for (int i = 0; i < size; i++) {
        strcat(result, arr[i]);
        strcat(result, " ");
    }
    return result;
}

char* printArrDouble(double *arr, int size) {
    static char result[1000] = "";
    result[0] = '\0';
    for (int i = 0; i < size; i++) {
        char temp[20];
        sprintf(temp, "%.1f ", arr[i]);
        strcat(result, temp);
    }
    return result;
}

int main() {
    long n, m;
    int t = 1;
    while (t--) {
        scanf("%ld %ld", &n, &m);
        if (n == 0) {
            printf("0\n");
            continue;
        }
        if (n == m) {
            printf("0\n");
            continue;
        }
        if (n > m) {
            n = m + n % m;
            n -= m;
            printf("%ld\n", n < m - n ? n : m - n);
            continue;
        }
        printf("%ld\n", n < m - n ? n : m - n);
    }
    return 0;
}
// End of Code
